src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -667,6 +668,40 @@ public override ImmutableArray<string> TupleElementNames
         }
 
         /// <summary>
+        /// Returns the symbols for all the tuple fields (in order and cached).
+        /// </summary>
+        public override ImmutableArray<FieldSymbol> TupleElementFields
+        {
+            get
+            {
+                if (_lazyFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyFields, CreateFields());
+                }
+
+                return _lazyFields;
+            }
+        }
+
+        private ImmutableArray<FieldSymbol> CreateFields()
```

Comment:
The name CreateFields is misleading. We are not creating fields, we are collecting them. CollectTupleElementFields?